File: tests/subscriber.js
const nsq = require('nsqjs')

const reader = new nsq.Reader('skripsi', 'pdf2', {
  lookupdHTTPAddresses: '127.0.0.1:4161'
})

reader.connect()

reader.on('message', msg => {
  console.log('Received message [%s]: %s', msg.id, msg.body.toString())
  msg.finish()
})

File: tests/subscriber.py
import nsq


def handler(message):
    print(message.body)
    return True


r = nsq.Reader(message_handler=handler,
               lookupd_http_addresses=['http://localhost:4161'],
               topic='skripsi', channel='method5', lookupd_poll_interval=15)
nsq.run()
